webapp/packages/plugin-settings-panel/src/SettingsPanel/SettingsList.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { observer } from 'mobx-react-lite';
+import { useEffect, useRef } from 'react';
+
+import { Container, Form, Group, GroupTitle, useTranslate } from '@cloudbeaver/core-blocks';
+import { type ISettingDescriptionWithScope, ROOT_SETTINGS_GROUP, type SettingsGroup } from '@cloudbeaver/core-plugin';
+import type { ISettingsSource } from '@cloudbeaver/core-settings';
+import type { ITreeData } from '@cloudbeaver/plugin-navigation-tree';
+
+import { getSettingGroupId } from './getSettingGroupId';
+import { getSettingGroupIdFromElementId } from './getSettingGroupIdFromElementId';
+import { querySettingsGroups } from './querySettingsGroups';
+import { Setting } from './Setting';
+
+interface Props {
+  treeData: ITreeData;
+  source: ISettingsSource;
+  settings: Map<SettingsGroup, ISettingDescriptionWithScope<any>[]>;
+  onSettingsOpen?: (groupId: string) => void;
+}
+
+export const SettingsList = observer<Props>(function SettingsList({ treeData, source, settings, onSettingsOpen }) {
+  const ref = useRef<HTMLDivElement>(null);
+  const translate = useTranslate();
+  const list = [];
+  const groups = [...treeData.getChildren(treeData.rootId)];
+
+  while (groups.length) {
+    const groupId = groups[0];
+    groups.splice(0, 1, ...treeData.getChildren(groupId));
+
+    const group = ROOT_SETTINGS_GROUP.get(groupId)!;
+    const groupSettings = settings.get(group);
+
+    list.push({ group, settings: groupSettings || [] });
+  }
+
+  useEffect(() => {
```

Comment:
I would like to see this logic moved to the hook so it is easier to read this component

Reply:
fixed